controllers/containerimagebuild_resources.go

```diff
@@ -196,26 +133,51 @@ func (r *ContainerImageBuildReconciler) jobForBuild(cib *forgev1alpha1.Container
 	for k, v := range r.JobConfig.Labels {
 		podMeta.Labels[k] = v
 	}
+	for k, v := range cib.Annotations {
+		podMeta.Annotations[k] = v
+	}
 	for k, v := range r.JobConfig.Annotations {
 		podMeta.Annotations[k] = v
 	}
 
 	// setup security context
+	podSecCtx := &corev1.PodSecurityContext{
+		FSGroup: pointer.Int64Ptr(1000),
+	}
 	secCtx := &corev1.SecurityContext{
 		RunAsUser: pointer.Int64Ptr(1000),
+		SELinuxOptions: &corev1.SELinuxOptions{
+			// TODO: this is currently required, because the default container SELinux rules
+			// do not seem to allow the remount,ro system calls that containerd uses. "spc_t"
+			// is a "special, super-privileged container" type (https://danwalsh.livejournal.com/74754.html)
+			Type: "spc_t",
+		},
 	}
 	if r.JobConfig.GrantFullPrivilege {
+		podSecCtx.FSGroup = nil
 		secCtx.RunAsUser = pointer.Int64Ptr(0)
 		secCtx.Privileged = pointer.BoolPtr(true)
 	}
 
 	// setup volumes and mounts used by main container
-	var volumes []corev1.Volume
+	volumes := []corev1.Volume{
+		{
+			Name: "state-dir",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+	}
```

Comment:
Can you help me understand why we started using an `emptyDir`?

Reply:
It ended up being  mostly because of moby/buildkit#1634, there were weird, different semantics when mounting overlay on overlay. I do think it's actually a good idea in practice anyway, since you're guaranteed a more "known" filesystem?